src/board_controller/aavaa/aavaa_v3.cpp

```diff
@@ -0,0 +1,563 @@
+                data[0], data[1], data[2], data[3], data[4]);
+        }
+        return;
+    }
+
+    int num_rows = board_descr["default"]["num_rows"];
+
+    std::vector<int> eeg_channels = board_descr["default"]["eeg_channels"];
+
+    if (size == 244)
+    {
+        data += 3;
+        size -= 3;
+        Incoming_BLE_Data_Buffer.insert (Incoming_BLE_Data_Buffer.end (), data, data + size);
+    }
+    else if (size > 1)
+    {
+        ++data;
+        --size;
+        Incoming_BLE_Data_Buffer.insert (Incoming_BLE_Data_Buffer.end (), data, data + size);
+    }
+
+    while (Incoming_BLE_Data_Buffer.size () >= SIZE_OF_DATA_FRAME)
+    {
+        if (Incoming_BLE_Data_Buffer[0] != START_BYTE)
+        {
+            Incoming_BLE_Data_Buffer.pop_front (); // discard data, we have half a frame here
+            continue;
+        }
+
+        uint8_t *data_frame = new uint8_t[SIZE_OF_DATA_FRAME];
+        for (int i = 0; i < SIZE_OF_DATA_FRAME; ++i)
+        {
+            data_frame[i] = Incoming_BLE_Data_Buffer.front ();
+            Incoming_BLE_Data_Buffer.pop_front ();
+        }
+
+        if (data_frame[SIZE_OF_DATA_FRAME - 1] != END_BYTE)
+        {
+            safe_logger (
+                spdlog::level::warn, "Wrong End Byte: {}", data_frame[SIZE_OF_DATA_FRAME - 1]);
+            continue;
+        }
+
+        double *package = new double[num_rows];
+        for (int i = 0; i < num_rows; i++)
+        {
+            package[i] = 0.0;
+        }
+
+        // package num
+        package[board_descr["default"]["package_num_channel"].get<int> ()] = (double)data_frame[1];
+
+        // eeg
+        for (unsigned int i = 0; i < eeg_channels.size (); i++)
+        {
+            // all the eeg_channels are 4 bytes and we skip the START byte and package number byte
+            float f_value;
+            std::memcpy (&f_value, data_frame + 2 + 4 * i, sizeof (f_value));
+            double d_value = static_cast<double> (f_value);
+            package[eeg_channels[i]] = EEG_SCALE * d_value;
+        }
+
+        package[board_descr["default"]["rotation_channels"][0].get<int> ()] =
+            IMU_SCALE * cast_16bit_to_int32 (data_frame + 34);
+        package[board_descr["default"]["rotation_channels"][1].get<int> ()] =
+            IMU_SCALE * cast_16bit_to_int32 (data_frame + 36);
+        package[board_descr["default"]["rotation_channels"][2].get<int> ()] =
+            IMU_SCALE * cast_16bit_to_int32 (data_frame + 38);
+
+        // battery byte
+        package[board_descr["default"]["battery_channel"].get<int> ()] = (double)data_frame[40];
+
+        // imu status byte
+        package[board_descr["default"]["rotation_calib_channel"].get<int> ()] =
+            (double)data_frame[41];
+
+        // timestamp
+        try
+        {
+            package[board_descr["default"]["other_channels"][0].get<int> ()] =
+                *reinterpret_cast<uint32_t *> (data_frame + 42) *
+                TIMESTAMP_SCALE; // get_timestamp ();
```

Comment:
is it a timestamp? If so why is it in other_channels?

Reply:
This is a board clock info channel for debug purposes, it is different from the `timestamp` channel that stores the timestamps when data is pushed in.